src/event_loop.rs

```diff
@@ -48,6 +50,17 @@ pub struct EventLoopWindowTarget<T: 'static> {
     pub(crate) _marker: PhantomData<*mut ()>, // Not Send nor Sync
 }
 
+/// Target that provides a handle to the display powering the event loop.
+///
+/// This type allows one to take advantage of the display's capabilities, such as
+/// querying the display's resolution or DPI, without having to create a window. It
+/// implements `HasDisplayHandle`.
+#[derive(Clone)]
+pub struct OwnedDisplayHandle {
```

Comment:
I don't really understand the purpose of this type. The `EventLoop` itself is what should have the `HasDisplayHandle` impl, and its lifetime is what should be used. The fact that you're not using this type in examples at all suggests me that there's no clear case when it should be used(at least you should use it inside the examples).

The `EventLoopWindowTarget` must have the same lifetime as the `EventLoop` itself, because it's simply a field on `EventLoop` in all the backends.

The `Clone` is also sort of questionable, because it simply will discard the lifetime attached to it, given that there's only a way to get `&OwnedDisplayHandle` not the owned type on its own?

if that's intent on `lifetime` casting, different method should be used with `unsafe` bit on it(because it's simply `mem::tranmsute` like thingy of donig `'static` lifetime).

Reply:
The reason why this type exists is for the `glutin` case. The scenario is that you need an object that has a display handle before a window is created, since you need to query the display. You can't use `EventLoop`, `&EventLoop` or a `DisplayHandle<'_>` taken from an `EventLoop` because the `EventLoop` is owned/borrowed mutably for event handling. You can't use the `EventLoopWindowTarget` that is provided in the event handler, since it disappears between events, which means it can't be used persistently. Normally I'd use a `Window`, but for `glutin` (outside of Win32) you haven't created a `Window` yet.

Therefore I created this type to fill that hole. Something that implements `HasDisplayHandle` that can be used according to Rust's borrowing system. This way, it can be used in the display position without any unsafe hacks.

If the `Clone` is a deal breaker I can get rid of it.

Reply:
>actually, it might be necessary for the borrowing rules to check out in this case.

I just think that it's `unsafe`, so you might have different method like `to_static`, which casts away lifetime.

Reply:
I have added documentation clarifying the purpose of `OwnedWindowHandle` and why it is safe.